### PR TITLE
Improve JsonRPC plugin settings

### PR DESCRIPTION
**Pull request by Garulf**
# Problem

- JsonRPC plugins are unable to make settings changes unless they have a `SettingsTemplate.yaml` file.
- Plugin's are unable to initiate new setting keys
- Plugin settings are only saved to disk upon exit

#  Solution

- [X] Allow plugins without `SettingsTemplate.yaml` to change their settings
- [X] Remove logic restricting settings key creation
- [X] Save plugin settings to file when updated

### CODE REVIEW

File: Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs [hunk 1]
```diff
@@ -81,6 +83,7 @@ public void UpdateSettings(IReadOnlyDictionary<string, object> settings)
                             break;
                     }
                 }
+                Save();
```
Comment (taooceros) [inline Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs:86]: maybe put this out of the loop. No need to save at every setting change?

Comment (Garulf) [reply Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs:86]: Oh whoops! 

I meant to make sure it was after the loop too!

File: Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs [hunk 2]
```diff
@@ -83,8 +83,8 @@ public void UpdateSettings(IReadOnlyDictionary<string, object> settings)
                             break;
                     }
                 }
-                Save();
             }
+            Save();
```
Comment (taooceros) [inline Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs:88]: sorry another thoughts that why didn't we keep this consistent with other plugins. `Save` only be called when manually triggered or flow being closed?

Comment (Garulf) [reply Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs:88]: I guess just personal preference. It feels off to not save to file on change.

Comment (taooceros) [reply Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs:88]: yeah ok

**Review by taooceros — APPROVED** [review]

**Pull request merged**